audio-worklet/src/bitcrusher-controller.js

```diff
@@ -30,112 +30,112 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-	 * @param  {Number} gain the volume of the output
-	 */
-	constructor(bitDepth, reduction, gain) {
-		this.context = new AudioContext();
-		this.bitDepth_ = bitDepth || 24;
-		this.reduction_ = reduction || 1;
-		
-		this.gainNode_ = new GainNode(this.context);
-		this.gainNode_.gain.value = gain || 0.5;
-	}
+  /**
+   * Initalizes and modifies bitcrusher_ in response to GUI input
+   * @param  {Number} bitDepth the number of bits for each sample
+   * @param  {Number} reduction the amount of sample rate reduction to apply
+   * @param  {Number} gain the volume of the output
+   */
+  constructor(bitDepth, reduction, gain) {
+    this.context = new AudioContext();
+    this.bitDepth_ = bitDepth || 24;
+    this.reduction_ = reduction || 1;
+    
+    this.gainNode_ = new GainNode(this.context);
+    this.gainNode_.gain.value = gain || 0.5;
+  }
 
-	/**
-	 * Initalize HTML elements when document has loaded
-	 * @param  {String} containerId the id of parent container
-	 */
-	initializeGUI(containerId) {
-		this.sourceButton_ = new SourceController(containerId, this.start.bind(this), 
-			this.stop.bind(this));
-		this.sourceButton_.disable();
-		
-		// Place 3 parameters in container to real-time adjust Bitcrusher_ settings
-		this.bitDepthParam_ = new ParamController(containerId, 
-			this.setBitDepth.bind(this), {type: "range", min: 1, max: 24,
-			step: 0.1, default: 24, name: "Bit Depth"});
-		this.bitDepthParam_.disable();
+  /**
+   * Initalize HTML elements when document has loaded
+   * @param  {String} containerId the id of parent container
+   */
+  initializeGUI(containerId) {
+    this.sourceButton_ = new SourceController(containerId, this.start.bind(this), 
+      this.stop.bind(this));
+    this.sourceButton_.disable();
+    
+    // Place 3 parameters in container to real-time adjust Bitcrusher_ settings
+    this.bitDepthParam_ = new ParamController(containerId, 
+      this.setBitDepth.bind(this), {type: "range", min: 1, max: 24,
+      step: 0.1, default: 24, name: "Bit Depth"});
+    this.bitDepthParam_.disable();
 
-		this.reductionParam_ = new ParamController(containerId, 
-			this.setReduction.bind(this), {type: "range", min: 1, max: 20,
-			step: 0.1, default: 1, name: "Sample Rate Reduction"});
-		this.reductionParam_.disable();
+    this.reductionParam_ = new ParamController(containerId, 
+      this.setReduction.bind(this), {type: "range", min: 1, max: 20,
+      step: 0.1, default: 1, name: "Sample Rate Reduction"});
+    this.reductionParam_.disable();
 
-		this.gainParam = new ParamController(containerId, 
-			this.setGain.bind(this), {type: "range", min: 0, max: 1,
-			step: 0.01, default: 0.5, name: "Volume"});
+    this.gainParam = new ParamController(containerId, 
+      this.setGain.bind(this), {type: "range", min: 0, max: 1,
+      step: 0.01, default: 0.5, name: "Volume"});
 
-		const songUrl = "audio/it-from-bit.mp3";
-		let that = this;
+    const songUrl = "audio/it-from-bit.mp3";
+    let that = this;
```

Comment:
Self? I'm creating a copy of "this" since the this referring to bitcrusher-controller will be replaced within the inner most scope of the fetch promise.